projects/questionset-editor-library/src/lib/components/question/question.component.ts

```diff
@@ -243,6 +244,31 @@ export class QuestionComponent implements OnInit, AfterViewInit, OnDestroy {
                   this.editorState.responseDeclaration = _.get(this.questionMetaData, 'responseDeclaration');
                 }
               }
+
+              if (this.questionInteractionType === 'match') {
```

Comment:
Lots of lines inside this if condition is similar to 'choice' condition. Due to this code duplication is happening. Please generalise this code.

Reply:
Completed